DSpace/dspace-angular
# Fix Angular SSR menu issues

## References
* Fixes #2503

## Description
This PR fixes both the menu duplication issue when a `''` child route is present as the issue with menus not rendering `OnClickMenuItemModel` correctly for route dependent menus.
For more information refer to the issue #2503.

## Instructions for Reviewers

Add an extra menu similarly to the explanation in the issue:

1. Add a menu section for a button in the `ItemPageRoutingModule` to the `''` route. Leave the statistics menu as is. This will result in the following route segment:
```           
          {
            path: '',
            component: ThemedItemPageComponent,
            pathMatch: 'full',
            data: {
              menu: {
                public: [{
                  id: 'function-test',
                  active: false,
                  visible: true,
                  model: {
                    type: MenuItemType.ONCLICK,
                    text: 'Click me and verify a console log is printed.',
                    function: () => console.log('Clicking the button worked!')
                  } as OnClickMenuItemModel,
                  icon: 'hand-paper',
                  index: 3,
                }],
              },
            }
          }

```
2. Build the DSpace in production mode
3. Go to an item page
4. Open the console log in your browser and hard refresh the page
5. Click the entry in the navbar menu saying "Click me and verify a console log is printed."
6. Verify that no error shows up but that instead the following message is shown: "Clicking the button worked!" 

## Checklist
